### DISCUSSION TITLE
Has anyone figured out how to do page transitions with <AnimatePresence> in Next 13.

### DISCUSSION THREAD

**Discussion by williamlmao** [motiondivision/motion#1775] [Q&A]
I'm not sure if this is even possible, but I'm trying to recreate [page transitions](https://wallis.dev/blog/nextjs-page-transitions-with-framer-motion) in Next 13. 

I've tried wrapping <AnimatePresence> around {children} inside of /app/layout.ts, but that doesn't seem to work. I can't really put <AnimatePresence> in my /app/page.ts because it is a server component and I have to keep it that way because of async child components. 

Would love to hear if anyone has found other solutions for this! Thanks.

**Comment by mosesoak** [comment]
@williamlmao Yes we're doing this, see https://github.com/framer/motion/discussions/1771

This should work in other frameworks besides Next as well, just keep in mind that _app is a top-level page wrapper in Next.

1. `<AnimatePresence>` in `_app` without any special `mode` set on it
2. Which wraps a per-page Layout component with a `key` set on it that changes based on route `<Layout key={router.asPath}>{children}</Layout>`
3. The Layout's outer container needs to be set to `position: absolute` and then have z-indexes set based on the transition state. This can be done using variants. For example, in initial zIndex: 0, in animate zIndex: 1, in exit zIndex 2 -- or switch these around to reverse the overlap.
4. The actual visual transition can then just be run inside the Layout component. For example, animating a `clipPath` based mask.

Using this technique we're able to overlap pages. The post I mentioned above is asking whether there's any standard way to reset scroll during this kind of transition, something we solved on our own.

Hope this helps you and others! ✨

**Reply by williamlmao** [reply]
@mosesoak thanks for writing back! is your response still referring to Next 12? I think the _app.ts is a Next 12 thing, in next 13 theres /app/page.tsx, but it turns out that is no longer the root. /app/layout.tsx is the root because it wraps /app/page.tsx.

**Reply by dvartic** [reply]
I am facing the same issue. As per Next13 doc, I use Template with a motion component. Template should receive a key prop automatically. According to the same docs, Template is placed just after the last element in Layout, so I place AnimatePresence as that last element. However, it doesn't work. 

Did you find a solution?

**Comment by RobinVanvoorden** [comment]
I'm having the same issues. I tried your solution, but exit animations still seem broken...
Here's a sandbox of my attempt, maybe I've missed something: [CodeSandBox Link](https://codesandbox.io/p/github/Neomas/next-js-framer-motion/main?file=%2Fpages%2F_app.tsx&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A15%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A15%7D%5D&workspace=%257B%2522activeFileId%2522%253A%2522cleyadrp1002xg8f6gkpyh71f%2522%252C%2522openFiles%2522%253A%255B%2522%252FREADME.md%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522spaces%2522%253A%257B%2522cleyadu5e001d3b6if50lt6ic%2522%253A%257B%2522key%2522%253A%2522cleyadu5e001d3b6if50lt6ic%2522%252C%2522name%2522%253A%2522Default%2522%252C%2522devtools%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522key%2522%253A%2522cleyb1l6j00b83b6iw4nfwiax%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522key%2522%253A%2522cleyb1hvh00733b6i06azc8jc%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D%257D%252C%2522currentSpace%2522%253A%2522cleyadu5e001d3b6if50lt6ic%2522%252C%2522spacesOrder%2522%253A%255B%2522cleyadu5e001d3b6if50lt6ic%2522%255D%252C%2522hideCodeEditor%2522%253Afalse%257D)

**Comment by danielfoxp2** [comment]
Has you guys found a way?

**Reply by RobinVanvoorden** [reply]
Yes, but the issue seems to lie with Next and not with Framer motion. There is an issue open about it on their git [https://github.com/vercel/next.js/issues/17464](url). At the bottom there's also a link to a package that should fix the issue. I haven't tried that one myself, but it seems promising.

Good luck!

**Reply by alexeigs** [reply]
No longer available.
